Platon/backend/app/search_engine/forms.py

```diff
@@ -34,4 +34,21 @@ class WorkspaceSearchForm(Form):
 ws_search_parser.add_argument('event_filter',type=str,help="Give Acronym",location='args')
 ws_search_parser.add_argument('page',type=int,help="Page ID(0-Indexed)",location='args')
 ws_search_parser.add_argument('per_page',type=int,help="Number of Records in a Page",location='args')
-ws_search_parser.add_argument('auth_token', type=str,help="Authantication Token(If registered)",location='headers')
\ No newline at end of file
+ws_search_parser.add_argument('auth_token', type=str,help="Authantication Token(If registered)",location='headers')
+
+class UpcomingEventsSearchForm(Form):
+    search_query = StringField("search_query",validators=[validators.DataRequired()])
+    date_filter = StringField("date_filter")
+    deadline_filter = StringField("deadline_filter")
```

Comment:
Why these 2 are strings and how do you filter these fields by only using 1 value? It is not user friendly to filter the dates by only using 1 value. I think it will be better to use 2 date values to represent an interval for date filtering.
Also, I can not understand why these 2 are StringFileds. I think you should take input by using DateTimeFields to make the format of dates uniform.